src/config/providers/k8s/agones.rs

```diff
@@ -284,8 +284,10 @@ where
     T: Default + Deserialize<'de>,
     D: serde::de::Deserializer<'de>,
 {
-    let opt = Option::deserialize(deserializer)?;
-    Ok(opt.unwrap_or_default())
+    Ok(match Option::deserialize(deserializer) {
+        Ok(opt) => opt.unwrap_or_default(),
+        Err(_) => T::default(),
```

Comment:
We probably only want to fail if the field is missing, but may want to capture other errors.

Although serde doesn't give us a nice way to to do that other than check the error text for "missing field" (afaik).

https://docs.rs/serde/latest/serde/de/trait.Error.html#method.missing_field

https://github.com/serde-rs/serde/blob/3202a6858a2802b5aba2fa5cf3ec8f203408db74/serde/src/de/mod.rs#L287

Is it worth doing the extra for checking for this specific error type, and letting others through - or should we always fall back to default on any error silently?